solr/core/src/java/org/apache/solr/handler/export/ExportWriter.java

```diff
@@ -828,4 +829,60 @@ public String getMessage() {
       return "Early Client Disconnect";
     }
   }
+
+  /**
+   * Creates a complete field list using the provided field list by expanding any glob patterns into
+   * field names
+   *
+   * @param fields the original set of fields provided
+   * @param searcher an index searcher to access schema info
+   * @return a complete list of fields included any fields matching glob patterns
+   * @throws IOException if a provided field does not exist or cannot be retrieved from the schema
+   *     info
+   */
+  private List<SchemaField> expandFieldList(String[] fields, SolrIndexSearcher searcher)
+      throws IOException {
+    List<SchemaField> expandedFields = new ArrayList<>(fields.length);
+    Set<String> fieldsProcessed = new HashSet<>();
+    for (String field : fields) {
+      try {
+        if (field.contains("*")) {
+          expandedFields.addAll(getGlobFields(field, searcher, fieldsProcessed));
+        } else {
+          if (fieldsProcessed.add(field)) {
+            expandedFields.add(searcher.getSchema().getField(field));
+          }
+        }
+      } catch (Exception e) {
+        throw new IOException(e);
+      }
+    }
+
+    return new ArrayList<>(expandedFields);
+  }
+
+  /**
+   * Create a list of schema fields that match a given glob pattern
+   *
+   * @param fieldPattern the glob pattern to match
+   * @param searcher an index search to access schema info
+   * @param fieldsProcessed the set of field names already processed to avoid duplicating
+   * @return a list of fields that match a given glob pattern
+   */
+  private List<SchemaField> getGlobFields(
+      String fieldPattern, SolrIndexSearcher searcher, Set<String> fieldsProcessed) {
+    List<SchemaField> schemaFields = new ArrayList<>();
```

Comment:
nit (please feel free to ignore!):

Why not just make this method void and pass `expandedFields` as an arg to be populated? Avoid creating a new list and iterating it -- and we're already doing similar with `fieldsProcessed` ...